src/filesystem.cpp

```diff
@@ -70,16 +171,21 @@ void FileSystemAccess::escapefsincompatible(string* name) const
 
     char buf[4];
     unsigned char c;
+    int fileSystemType = getlocalfstype(dstPath);
 
     // replace all occurrences of a badchar with %xx
     for (size_t i = name->size(); i--; )
     {
         c = (unsigned char)(*name)[i];
 
-        if (!islocalfscompatible(c))
+        if (!islocalfscompatible(c, fileSystemType))
```

Comment:
I think we need to take utf-8 encoding into account here - eg. if a utf-8 character is encoded across two bytes, the second byte could match the char "/", but overall that two-byte character should not be changed.

Reply:
You are right this need to be adjusted, thanks so much for the comments ;)

Reply:
@mattw-mega could you have a look to the latest commits, I think that solves the problem with multi-by characters: